Add dev_build and dev_mode flags to SCons compilation instructions for desktop editor pages (#10019)
godotengine/godot-docs: contributing/development/compiling/compiling_for_linuxbsd.rst

    ``linuxbsd``. If you are looking to compile Godot 3.x, make sure to use the
    `3.x branch of this documentation <https://docs.godotengine.org/en/3.6/development/compiling/compiling_for_x11.html>`__.

> If you wish to add and use breakpoints while developing, add dev_build=yes

Reviewer: ```suggestion
If you wish to add and use breakpoints while developing, add ``dev_build=yes``
```

Reviewer 2: `dev_build` enable additional runtime checks and disables optimization, it will make debugging more precise, but is not strictly required to debug/set brakepoints (with `debug_symbols=yes` it can be done in optimized build as well).

`dev_mode` is not related to debugging at all, it's enabling extra check during engine build.

> and dev_mode=yes to the command::

Reviewer: ```suggestion
and ``dev_mode=yes`` to the command::
```

    ``linuxbsd``. If you are looking to compile Godot 3.x, make sure to use the
    `3.x branch of this documentation <https://docs.godotengine.org/en/3.6/development/compiling/compiling_for_x11.html>`__.

.. tip::
    If you are compiling Godot to make changes or contribute to the engine,
    you may want to use the SCons options ``dev_build=yes`` or ``dev_mode=yes``. 

>     See `Development and Production Aliases <https://docs.godotengine.org/en/stable/contributing/development/compiling/introduction_to_the_buildsystem.html#development-and-production-aliases>`__

Reviewer 3: ```suggestion
    See :ref:`doc_introduction_to_the_buildsystem_development_and_production_aliases`
```
We can use internal links here.

You'll also have to add an anchor to the section header in `Introduction to the buildsystem`. Note that there is an underscore `_` in the anchor definition but not the reference.

```diff
- Development and production aliases
- ----------------------------------
+ .. _doc_introduction_to_the_buildsystem_development_and_production_aliases:
+
+ Development and production aliases
+ ----------------------------------
```

The same changes need to be applied to the other two tip boxes.
